packages/messageport/tizen/src/messageport_tizen_plugin.cc

```diff
@@ -9,25 +9,117 @@
 #include <flutter/event_stream_handler_functions.h>
 #include <flutter/method_channel.h>
 #include <flutter/plugin_registrar.h>
+#include <flutter/standard_message_codec.h>
 #include <flutter/standard_method_codec.h>
 
-#include <map>
-#include <memory>
+#include <set>
 #include <sstream>
 #include <string>
 
 #include "log.h"
 #include "messageport.h"
 
+typedef flutter::EventChannel<flutter::EncodableValue> FlEventChannel;
+typedef flutter::EventSink<flutter::EncodableValue> FlEventSink;
+typedef flutter::MethodCall<flutter::EncodableValue> FlMethodCall;
+typedef flutter::MethodChannel<flutter::EncodableValue> FlMethodChannel;
+typedef flutter::MethodResult<flutter::EncodableValue> FlMethodResult;
+typedef flutter::StreamHandler<flutter::EncodableValue> FlStreamHandler;
+typedef flutter::StreamHandlerError<flutter::EncodableValue>
+    FlStreamHandlerError;
+
 namespace {
 
+template <typename T>
+bool GetValueFromEncodableMap(const flutter::EncodableMap *map, const char *key,
+                              T &out) {
+  auto iter = map->find(flutter::EncodableValue(key));
+  if (iter != map->end() && !iter->second.IsNull()) {
+    if (auto *value = std::get_if<T>(&iter->second)) {
+      out = *value;
+      return true;
+    }
+  }
+  return false;
+}
+
+class LocalPortStreamHandlerHandlerError : public FlStreamHandlerError {
+ public:
+  LocalPortStreamHandlerHandlerError(
+      const std::string &error_code, const std::string &error_message,
+      const flutter::EncodableValue *error_details)
+      : error_code_(error_code),
+        error_message_(error_message),
+        FlStreamHandlerError(error_code_, error_message_, error_details) {}
+
+ private:
+  std::string error_code_;
+  std::string error_message_;
+};
+
+class LocalPortStreamHandler : public FlStreamHandler {
+ public:
+  LocalPortStreamHandler(LocalPort *local_port) : local_port_(local_port) {}
+
+  std::unique_ptr<FlStreamHandlerError> OnListenInternal(
+      const flutter::EncodableValue *arguments,
+      std::unique_ptr<FlEventSink> &&events) override {
+    event_sink_ = std::move(events);
+    std::optional<MessagePortError> error =
+        local_port_->Register([this](const Message &message) {
+          if (message.error.size()) {
+            event_sink_->Error(message.error);
+            return;
+          }
+
+          flutter::EncodableMap map;
+          auto value =
+              flutter::StandardMessageCodec::GetInstance().DecodeMessage(
+                  *message.encoded_message.get());
+          map[flutter::EncodableValue("message")] = *(value.get());
+          if (message.remote_app_id.size()) {
+            map[flutter::EncodableValue("remoteAppId")] =
+                flutter::EncodableValue(message.remote_app_id);
+          }
+          if (message.remote_port.size()) {
+            map[flutter::EncodableValue("remotePort")] =
+                flutter::EncodableValue(message.remote_port);
+          }
```

Comment:
`remoteAppId` and `remotePort` can't be optional. Non-null string values are expected by the Dart side.

Reply:
No. They are expected to be optional.

https://github.com/flutter-tizen/plugins/blob/882cf4339cfab3d6788c69cf8186b6505573487c/packages/messageport/lib/messageport_tizen.dart#L53

Reply:
You're right. We shouldn't necessarily mark `remote_app_id` and `remote_port` as `std::optional` and can just use `""` if we're sure that they cannot be empty strings.

Reply:
If we use "' for remote port, the final result of the Dart API will be different. As a result, we cannot pass tests expecting remort port to be null.

Reply:
Uhm, what I meant was, the current C++ implementation which uses `std::string("")` to represent null strings should be okay.